infrastructures/infrastructure-gce/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/GCEInfrastructure.java

```diff
@@ -322,19 +322,23 @@ private void deployInstancesWithFullNodes(int nbInstancesToDeploy) {
 
         connectorIaasController.waitForConnectorIaasToBeUP();
 
-        createInfrastructureIfNeeded();
+        String infrastructureId = getInfrastructureId();
+
+        createInfrastructureIfNeeded(infrastructureId);
 
         List<String> nodeStartCmds = buildNodeStartScripts(numberOfNodesPerInstance);
 
-        Set<String> instancesIds = createInstanceWithNodesStartCmd(nbInstancesToDeploy, nodeStartCmds);
+        Set<String> instancesIds = createInstanceWithNodesStartCmd(infrastructureId,
+                                                                   nbInstancesToDeploy,
+                                                                   nodeStartCmds);
 
         declareDeployingNodes(instancesIds, numberOfNodesPerInstance, nodeStartCmds.toString());
     }
 
-    private void createInfrastructureIfNeeded() {
+    private void createInfrastructureIfNeeded(String infrastructureId) {
         // Create infrastructure if it does not exist
         if (!isCreatedInfrastructure) {
-            connectorIaasController.createInfrastructure(getInfrastructureId(),
+            connectorIaasController.createInfrastructure(infrastructureId,
```

Comment:
I don't understand why you replace 'getInfrastructureId()' with 'infrastructureId'. Would you clarify please?

Reply:
I just thought it's not necessary to parse infrastructureId multiple times from nodeSource during acquireNodes.